google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/stub/metrics/OutstandingRpcsMetricTracker.java

```diff
@@ -0,0 +1,111 @@
+  private final AtomicReference<String> lbPolicyRef = new AtomicReference<>();
+
+  // Base attributes common to all recordings
+  private static final Attributes TRANSPORT_GRPC_ATTR =
+      Attributes.builder().put("transport_type", "grpc").build();
+
+  // Attributes for unary and streaming RPCs, built on demand in run()
+  @Nullable private Attributes unaryAttributes;
+  @Nullable private Attributes streamingAttributes;
+
+  public OutstandingRpcsMetricTracker(OpenTelemetry openTelemetry) {
+    Meter meter = openTelemetry.getMeter(METER_NAME);
+    this.outstandingRpcsHistogram =
+        meter
+            .histogramBuilder(OUTSTANDING_RPCS_PER_CHANNEL_NAME)
+            .ofLongs()
+            .setDescription(
+                "A distribution of the number of outstanding RPCs per connection in the client pool, sampled periodically.")
+            .setUnit("1")
+            .build();
+  }
+
+  /**
+   * Registers the provider for the channel pool entries. This should be called by the component
+   * that creates the BigtableChannelPool.
+   */
+  public void registerChannelInsightsProvider(BigtableChannelPoolObserver channelInsightsProvider) {
+    this.bigtableChannelInsightsProviderRef.set(channelInsightsProvider);
+  }
+
+  /** Register the current lb policy * */
+  public void registerLoadBalancingStrategy(String lbPolicy) {
+    this.lbPolicyRef.set(lbPolicy);
+  }
+
+  /** Starts the periodic collection. */
+  public ScheduledFuture<?> start(ScheduledExecutorService scheduler) {
+    return scheduler.scheduleAtFixedRate(
+        this, SAMPLING_PERIOD_SECONDS, SAMPLING_PERIOD_SECONDS, TimeUnit.SECONDS);
+  }
+
+  @Override
+  public void run() {
+    BigtableChannelPoolObserver channelInsightsProvider = bigtableChannelInsightsProviderRef.get();
+    if (channelInsightsProvider == null) {
+      return; // Not registered yet
+    }
+    String lbPolicy = lbPolicyRef.get();
+    if (lbPolicy == null) {
+      lbPolicy = "ROUND_ROBIN";
+    }
+
+    // Build attributes if they haven't been built yet or were invalidated
+    if (unaryAttributes == null || streamingAttributes == null) {
+      Attributes baseAttrs = TRANSPORT_GRPC_ATTR.toBuilder().put("lb_policy", lbPolicy).build();
+      this.unaryAttributes = baseAttrs.toBuilder().put("streaming", false).build();
+      this.streamingAttributes = baseAttrs.toBuilder().put("streaming", true).build();
+    }
+    List<? extends BigtableChannelObserver> channelInsights =
+        channelInsightsProvider.getChannelInfos();
+    if (channelInsights == null || channelInsights.isEmpty()) {
+      return;
+    }
+    for (BigtableChannelObserver info : channelInsights) {
+      long currentOutstandingUnaryRpcs = info.getOutstandingUnaryRpcs();
+      long currentOutstandingStreamingRpcs = info.getOutstandingStreamingRpcs();
+      // Record outstanding unary RPCs with streaming=false
+      outstandingRpcsHistogram.record(currentOutstandingUnaryRpcs, unaryAttributes);
```

Comment:
I'm having a hard time understanding how we'll interpret the histogram.. are we trying to do something similar to per connection error count, where if there's a big difference between P50 and P99, we know the error is more likely to be coming from one channel, and for this metric, if there's a big difference between P50 and P99, we know that the traffic is not evenly distributed?